### PR TITLE
News api integration

### PR DESCRIPTION
**Pull request by Chufle**
change-log:
- new lambda function news_crawler(), to import news data from news-api to kutter-table
- API endpoint for new_crawler():
    `https://***.execute-api.us-west-2.amazonaws.com/prod/news_crawler?topic=***&dateFrom=***`
to search for all news regarding a topic and from a date. For example all news from Hamburg from yesterday and today would be:
    `https://***.execute-api.us-west-2.amazonaws.com/prod/news_crawler?topic=Hamburg&dateFrom=2022-12-11`

Next Step:
- implement error handling and limiting number of news objects by a maximum value

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -26,4 +26,4 @@ jobs:
         run:  |
           cd infrastructure
           terraform init
-          terraform apply -auto-approve
\ No newline at end of file
+          terraform apply -auto-approve -var NEWS_API_KEY=${{ secrets.NEWS_API_KEY }}
```
Comment (irenepapaspyratos) [inline .github/workflows/deploy.yml:29]: Empty Line

**Review by irenepapaspyratos — COMMENTED** [review]
Only one missing line found. 👏

File: news_crawler/src/news_crawler.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+       'apiKey='+apiKey)
+    response = requests.get(url)
+    return response.json()
+
+def generate_db_object_id():
+    object_id = str(uuid.uuid4())
+    return object_id
+
+def map_news(news_from_api):
+    news = []
```
Comment (fabianschmauder) [inline news_crawler/src/news_crawler.py:27]: if you want to improve this. have a look at mapping with lambda

File: news_crawler/src/news_crawler.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "title": api_news["title"],
+            "creationDate": api_news["publishedAt"],
+            "url": api_news["url"],
+        })
+    return news
+
+def put_db_object_news(news_from_api):
+    for api_news in news_from_api["articles"]:
+        object_id = generate_db_object_id()
+        response = kutter_table.put_item(
```
Comment (fabianschmauder) [inline news_crawler/src/news_crawler.py:41]: there is also a batch write option

**Review by fabianschmauder — COMMENTED** [review]
Great.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (Chufle) [reply .github/workflows/deploy.yml]: ok :)

File: infrastructure/api_gateway.tf [hunk 4]
```diff
@@ -1,58 +1,114 @@
...
   description = "for AWS Lambda trigger"
 }
 
-resource "aws_api_gateway_resource" "proxy" {
-  rest_api_id = "${aws_api_gateway_rest_api.get-objects-API.id}"
-  parent_id   = "${aws_api_gateway_rest_api.get-objects-API.root_resource_id}"
-  path_part   = "{proxy+}"
+# get-object endpoint:
+
+resource "aws_api_gateway_resource" "get_object" {
```
[the remark this answers is not in the document]
Comment (Chufle) [reply infrastructure/api_gateway.tf:8]: good idea

File: get_object/src/get_object.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    response = table.get_item(
+        Key={
+            'objectId': objectId
+        }
+    )
+    return response['Item']
+
+def handler(event, context):
+    objectId = event ['queryStringParameters']['objectId']
+    item = get_object_data(objectId)
```
[the remark this answers is not in the document]
Comment (Chufle) [reply get_object/src/get_object.py:17]: have created an issue for that

**Pull request merged**